pkg/util/template.go

```diff
@@ -214,6 +220,38 @@ func configMapEntriesGenerator(
 	}
 }
 
+// lookup does a dot-separated path lookup on the input struct or map. If the input or any of
+// its children on the targeted path are not a map or struct, it returns nil.
+func lookup(path string, input interface{}) interface{} {
+	obj := reflect.ValueOf(input)
+
+	components := strings.Split(path, ".")
+
+	for i := 0; i < len(components); {
+		component := components[i]
+
+		switch obj.Kind() {
+		case reflect.Struct:
+			obj = obj.FieldByName(component)
+			i++
+		case reflect.Map:
+			obj = obj.MapIndex(reflect.ValueOf(component))
+			i++
+		case reflect.Ptr, reflect.Interface:
+			// Get the thing being pointed to or interfaced, don't advance index
+			obj = obj.Elem()
+		default:
+			// Got an unexpected type
+			return nil
```

Comment:
Would a panic be more appropriate here? It seems like we'd be using the function really wrong if we run into this case.

Alternatively, we could return the value itself.

Reply:
Yeah, so there are two cases here- one that's an error and one that isn't. Separated them and updated the function to return an error if the user tries to traverse a non-map type.